app/jobs/event_notification_job.rb

```diff
@@ -0,0 +1,13 @@
+class EventNotificationJob < ApplicationJob
+  def self.perform
+    Event.requiring_notification.each do |event|
+      event_date = event.date
```

Comment:
Unnecessary variable assignment. `event`  is still available in the nested loop and typing `event_date` doesn't save much space.